dcore/state.py

```diff
@@ -71,24 +77,44 @@ def dump(self):
             self.output.dumplist = self.fieldlist
 
         funcs = self.xn.split()
+        field_dict = {name: func for (name, func) in zip(self.fieldlist, funcs)}
         to_dump = []
-        for name, f in zip(self.fieldlist, funcs):
+        for name, f in field_dict.iteritems():
             if name in self.output.dumplist:
                 to_dump.append(f)
             f.rename(name=name)
 
-        dumpdir = path.join("results", self.output.dirname)
+        if self.output.steady_state_dump_err:
+            init_funcs = self.x_init.split()
+            for name, f, f_init in zip(self.fieldlist, funcs, init_funcs):
+                if name in self.output.dumplist:
+                    err = Function(f.function_space(), name=name+'err').assign(f-f_init)
+                    field_dict[name+"err"] = err
+                    self.diagnostics.register(name+"err")
```

Comment:
Note that you don't actually need to allocate a new function for this guy if you're only going to use it in the diagnostics.  It suffices to put `f - f_init` in `field_dict`.  Because when you compute the l2 error you'll end up inserting `assemble(sqrt(dot(f - f_init, f-f_init))*dx)` which works fine.  If you're also going to output it for viz, you can't do this trick though.